components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/internal/util/TemplateManagerHelper.java

```diff
@@ -322,26 +321,61 @@ public static void validateTemplatesProperties(RuleTemplate ruleTemplate) throws
     /**
      * Checks whether a given Template file has valid content.
      * Validation criteria : //todo: confirm validation criteria for templates
-     * - name
-     * - maximumInstances
-     * - maxNumberOfNodes
-     * - javascript
-     * - siddhiApps
-     * - properties //todo: validate whether all templated elements are referred as properties?
+     * - type
+     * - content
+     * - ExposedStreamDefinition
      *
-     * @param template Given Template object
+     * @param ruleTemplate Given Template object
      * @throws TemplateManagerException
      */
-    public static void validateTemplate(Template template) throws TemplateManagerException {
-        //todo: no need mostly.
-    }
+    public static void validateTemplate(RuleTemplate ruleTemplate) throws TemplateManagerException {
+        Collection<Template> templates = ruleTemplate.getTemplates();
+        /**
+         * Validation for the
+         *
+         * **/
+        if(ruleTemplate.getType().equals(TemplateManagerConstants.TEMPLATE)){
+            for (Template template : templates) {
+                if (template.getType().isEmpty()){
+                    throw new TemplateManagerException("Invalid template. Template type cannot be null in rule " +
+                            "template " +ruleTemplate.getUuid());
+                }
+                if (!template.getType().equals(TemplateManagerConstants.SIDDHI_APP_TEMPLATE_TYPE)|| !template.getType()
+                        .equals(TemplateManagerConstants.GADGET)||!template.getType().equals(TemplateManagerConstants
+                        .DASHBOARD)){
+                    throw new TemplateManagerException("Invalid template. Template type only can be 'siddhiApp'," +
+                            "'gadget' or " +
+                            "'dashboard'" +
+                            " in rule template "+ ruleTemplate.getUuid());
```

Comment:
I think we can get this into 2 lines

Reply:
This was changed by df3bc5a752f0dfcd03393c5f62409814722e720b